### PR TITLE
RNA-Seq expression summary statistics module update

### PR DESCRIPTION
**Pull request by ewafula**
#### What scientific question is your analysis addressing?
- Update the  `rna-seq-expression-summary-stats` module to compute `all cohorts` and `each cohort` expression summary statistics using different independent samples created by the `independent-samples`  module. 
- set minimum samples size for `cancer_group` to `n>=3`

#### What was your approach?
Summary statistics are calculated using `../independent-samples/results/independent-specimens.rnaseq.primary.tsv`  for `all cohorts` and `../independent-samples/results/independent-specimens.rnaseq.primary.eachcohort.tsv` for `each cohort`.

#### What GitHub issue does your pull request address?
- https://github.com/PediatricOpenTargets/ticket-tracker/issues/138
- https://github.com/PediatricOpenTargets/ticket-tracker/issues/178

#### Which areas should receive a particularly close look?
Check if the module updates are what is required

#### Is there anything that you want to discuss further?
Check the update to handle independent samples for `all cohorts` and `each cohort` differently as they are propagated form `lines 379 - 597` after loading the ISL files shown  below:
```
lines: 379 -383
# all cohorts independent sample table
rna_ac_idps_df <- read_tsv(
  file.path('..', 'independent-samples', 'results',
            'independent-specimens.rnaseq.primary.tsv'),
  col_types = cols(.default = col_guess()))

lines: 398 -402 
# each cohort independent sample table
rna_ec_idps_df <- read_tsv(
  file.path('..', 'independent-samples', 'results',
            'independent-specimens.rnaseq.primary.eachcohort.tsv'),
  col_types = cols(.default = col_guess()))

```
#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?
YES

#### What types of results are included (e.g., table, figure)?
- `results/cancer_group_all_cohorts_sample_metadata.tsv`
- `results/cancer_group_each_cohort_sample_metadata.tsv`
- `long_n_tpm_mean_sd_quantile_gene_wise_zscore.json`
- `long_n_tpm_mean_sd_quantile_gene_wise_zscore.tsv`
- `long_n_tpm_mean_sd_quantile_group_wise_zscore.json`
- `long_n_tpm_mean_sd_quantile_group_wise_zscore.tsv`

#### Reproducibility Checklist

- [x ] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [ ] This analysis has been added to continuous integration.

#### Documentation Checklist

- [x] This analysis module has a `README` and it is up to date.
- [ ] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [x] The analytical code is documented and contains comments.

### CODE REVIEW

File: analyses/rna-seq-expression-summary-stats/01-tpm-summary-stats.R [hunk 1]
```diff
@@ -376,24 +376,44 @@ stopifnot(identical(
 ))
 
 
-# independent sample table
-rna_idps_df <- read_tsv(
+# all cohorts independent sample table
+rna_ac_idps_df <- read_tsv(
+  file.path('..', 'independent-samples', 'results',
```
Comment (runjin326) [inline analyses/rna-seq-expression-summary-stats/01-tpm-summary-stats.R]:
```suggestion
  file.path('../..', 'data',
```

File: analyses/rna-seq-expression-summary-stats/01-tpm-summary-stats.R [hunk 2]
```diff
@@ -376,24 +376,44 @@ stopifnot(identical(
...
+
+# subset histology data frame to have all cohorts independent samples in tpm_df
+rna_ac_htl_df <- htl_df %>%
+  filter(Kids_First_Biospecimen_ID %in% colnames(tpm_df),
+         Kids_First_Biospecimen_ID %in% rna_ac_idps_kfbids)
+stopifnot(all(rna_ac_htl_df$Kids_First_Biospecimen_ID %in% colnames(tpm_df)))
+
+# each cohort independent sample table
+rna_ec_idps_df <- read_tsv(
   file.path('..', 'independent-samples', 'results',
```
Comment (runjin326) [inline analyses/rna-seq-expression-summary-stats/01-tpm-summary-stats.R]:
```suggestion
  file.path('../..', 'data',
```
We can change the file path now that we release independent samples

File: analyses/rna-seq-expression-summary-stats/README.md [hunk 3]
```diff
@@ -174,12 +174,13 @@ Input:
 - `../../data/gene-expression-rsem-tpm-collapsed.rds`
 - `../../data/histologies.tsv`
 - `../../data/ensg-hugo-rmtl-mapping.tsv`
+- `../independent-samples/results/independent-specimens.rnaseq.primary.tsv`
```
Comment (runjin326) [inline analyses/rna-seq-expression-summary-stats/README.md]:
```suggestion
- `../../data/results/independent-specimens.rnaseq.primary.tsv`
```
Directory for this and the one after can be changed as well.

**Review by runjin326 — APPROVED** [review]
All the changes look good to me! Just a couple places where we can call independent samples directly from `data`. 
I also looked at the meta differences - all the differences in each-cohort were from the addition of cancer_group N between 3-4, and selection of different biospecimens (since we have a new isl).
The differences in all-cohorts have sample # drop in particular cancer groups (as compared to each-cohort) which is also expected since now we are using the correct isl. 
Could you please also make a merge of the most up-to-date dev? Should be ready fo merge after that!

File: analyses/rna-seq-expression-summary-stats/README.md (hunk 3 above)
Comment (ewafula) [reply analyses/rna-seq-expression-summary-stats/README.md]: @runjin326, Sure, I'll update immediately and resubmit the PR

**Comment by ewafula** [comment]
Updated script to read independent samples directly from `data` of the `OpenPedCan` repo.

**Pull request merged**